### PR TITLE
Fix/#CSV Injection Vulnerability Fixes

### PR DESCRIPTION
**Pull request by bvenceslas**
https://www.loom.com/share/e3b746003e88412f8b5ffbddde670106

### CODE REVIEW

File: api/package.json [hunk 1]
```diff
@@ -62,7 +62,7 @@
     "rxjs": "^7.3.0",
     "strings-file": "^0.0.5",
     "swagger-ui-express": "^4.1.6",
-    "typeorm": "^0.2.37",
```
Comment (evereq) [inline api/package.json:65]: @bvenceslas I don't see any changes in package.json, can you please remove that from PR? Thank you!

File: api/src/formatters/csv.ts [hunk 2]
```diff
@@ -29,9 +29,32 @@ export const csvParser: Parser = async (data: string) => {
...
+  if(!isInjected) return str;
+
+  const sliceStr = str.slice(1);
+  return csvInjectionProtector(sliceStr);
+}
+
 export const csvExporter: Exporter = async (data: IntermediateTranslationFormat) => {
+  
+  // clear some characters
+  const protectedTerm = csvInjectionProtector(data.translations[0].term);
```
Comment (evereq) [inline api/src/formatters/csv.ts:48]: @bvenceslas can you explain the logic here? Why we are using only the first element of the "translations" array? What will happen with other elements? Or it's always just one element!? Don't you think we cut the processing of the array to a single element with your code changes?

File: api/package.json (hunk 1 above)
Comment (bvenceslas) [reply api/package.json:65]: will do

File: api/src/formatters/csv.ts [hunk 3]
```diff
@@ -29,9 +29,32 @@ export const csvParser: Parser = async (data: string) => {
...
+  const protectedTranslation = csvInjectionProtector(data.translations[0].translation);
+
+  const payload = [{
+    term: protectedTerm.replace('0x0d', ''),
```
Comment (evereq) [inline api/src/formatters/csv.ts:52]: @bvenceslas what is "0x0d"?  Why you did not add it into `csvInjectionProtector` function?

File: api/src/formatters/csv.ts (hunk 2 above)
Comment (bvenceslas) [reply api/src/formatters/csv.ts:48]: oh Yes!
let me rework on it

File: api/src/formatters/csv.ts [hunk 4]
```diff
@@ -29,9 +29,32 @@ export const csvParser: Parser = async (data: string) => {
   };
 };
 
+const csvInjectionProtector = (str: string) => {
+  const riskyChars = ["=", "+", "-", "@", ",", ";", "0", "1", "2", "3", "4", "5", "6", "7", "8", "9"];
```
Comment (evereq) [inline api/src/formatters/csv.ts:33]: @bvenceslas from where did you get this function logic? Is it from https://github.com/destromas1/csv-injection-protector? If so, why you did not use the NPM package? Also, can you please add tests (we have a collection of tests in Traduora) to test different attempts to inject bad sequences and see how it's sanitized? Think you can use some payloads from https://github.com/payloadbox/csv-injection-payloads and also please review https://github.com/joanbono/GOCiS for some more test cases etc

File: api/src/formatters/csv.ts (hunk 2 above)
Comment (evereq) [reply api/src/formatters/csv.ts:48]: Yes please rework and see if all working both in UI and also run Unit Tests (there was 1 or 2 tests failing before, so run tests BEFORe and AFTER you changes)

File: api/src/formatters/csv.ts (hunk 4 above)
Comment (bvenceslas) [reply api/src/formatters/csv.ts]: that package didn't work here, on other windows project using npm yes,
but not with yarn, it was crashing, so i thought about a logic and did.

Comment (evereq) [reply api/src/formatters/csv.ts]: OK, but if you used code (copy / paste with some modifications) from that package, please add comments into files you copied like we are doing for example in https://github.com/ever-co/ever-gauzy/blob/develop/packages/core/src/core/core.module.ts:

```
// Copyright (c) 2021-2022 Ever Co. LTD
// Modified code from https://github.com/destromas1/csv-injection-protector 
// Originally MIT Licensed
// - see https://github.com/destromas1/csv-injection-protector/blob/master/LICENSE
// - original code `Copyright (c) 2019 Shahjada Talukdar`;
```

Thank you!

File: api/src/config.ts [hunk 5]
```diff
@@ -51,7 +51,7 @@ export const config = {
       host: env.TR_DB_HOST || '127.0.0.1',
       port: parseInt(env.TR_DB_PORT, 10) || 3306,
       username: env.TR_DB_USER || 'root',
-      password: env.TR_DB_PASSWORD || 'root',
```
Comment (evereq) [inline api/src/config.ts]: @bvenceslas I think that will crush some things... such default password used only during development to be able to connect to local DB with some default password I think....

Comment (bvenceslas) [reply api/src/config.ts]: i've set it to default

File: api/package.json (hunk 1 above)
Comment (evereq) [reply api/package.json:65]: Hm, seems file is still there in PR :(

**Pull request merged**